appnav/src/main/kotlin/io/element/android/appnav/LoggedInFlowNode.kt

```diff
@@ -127,8 +131,18 @@ class LoggedInFlowNode @AssistedInject constructor(
     )
 
     private val verificationListener = object : SessionVerificationServiceListener {
-        override fun onIncomingSessionRequest(sessionVerificationRequestDetails: SessionVerificationRequestDetails) {
-            backstack.singleTop(NavTarget.IncomingVerificationRequest(sessionVerificationRequestDetails))
+        override fun onIncomingSessionRequest(verificationRequest: VerificationRequest.Incoming) {
+            // Without this launch the rendering and actual state of this Appyx node's children gets out of sync, resulting in a crash.
+            // This might be because this method is called back from Rust in a background thread.
+            MainScope().launch {
+                // Wait until the app is in foreground to display the incoming verification request
+                appNavigationStateService.appNavigationState.first { it.isInForeground }
```

Comment:
Maybe add a timeout?

Reply:
Maybe, but I'm not sure what a sensible value would be. Maybe instead of the callback we should try to check if the request hasn't been cancelled.